executor.go

```diff
@@ -121,6 +125,20 @@ func (s *executor) run(mainCtx context.Context,
 		return mainCtx.Err()
 	}
 
+	batcherErrChan = make(chan error, 1)
+	batcherCtx, cancel := context.WithCancel(mainCtx)
```

Comment:
There's no need for cancalling the batcher context as it's only used by the batcher. We could potentially cancel the main context though so that the executor also quits, but forwarding the error will result in the same behavior.

Reply:
Yeah the err would be forwarded to the error chan and executor would quit afterwards

Should I just directly use the `mainCtx` in the batcher itself?